crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -127,31 +129,164 @@ impl Extension for QueryLimitsChecker {
         let cfg = ctx
             .data::<ServiceConfig>()
             .expect("No service config provided in schema data");
-        if query.len() > cfg.limits.max_query_payload_size as usize {
-            metrics
-                .request_metrics
-                .query_payload_too_large_size
-                .observe(query.len() as f64);
-            info!(
-                query_id = %query_id,
-                session_id = %session_id,
-                error_code = code::BAD_USER_INPUT,
-                "Query payload is too large: {}",
-                query.len()
-            );
-
-            return Err(graphql_error(
-                code::BAD_USER_INPUT,
-                format!(
-                    "Query payload is too large. The maximum allowed is {} bytes",
-                    cfg.limits.max_query_payload_size
-                ),
-            ));
-        }
-
         // Document layout of the query
         let doc = next.run(ctx, query, variables).await?;
 
+        let mutation_node = doc
+            .operations
+            .iter()
+            .find(|(_, operation)| operation.node.ty == OperationType::Mutation);
+
+        let is_mutation = mutation_node.is_some();
+
+        // this is a map of type node position, (tx_bytes, signatures) bytes for all nodes of type
+        // executeTransactionBlock
+        let mut tx_bytes_sigs = BTreeMap::new();
+
+        // A few checks in case it is a mutation
+        if is_mutation {
```

Comment:
Can you use `if let Some((_, operation)) = &mutation_node {` as the outer test, instead of `is_mutation`? It means you shouldn't need the inner `if let Some(...)` or the `is_mutation` local.

Reply:
Thanks for the suggestion, I switched to it.